spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/inbound/HazelcastContinuousQueryMessageProducer.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import com.hazelcast.core.IMap;
+import com.hazelcast.query.SqlPredicate;
+
+import org.springframework.integration.hazelcast.listener.HazelcastEntryListener;
+
+/**
+ * Hazelcast Continuous Query Message Producer is a message producer which enables
+ * HazelcastEntryListener with a SqlPredicate in order to listen related distributed map
+ * events in the light of defined predicate and sends events to related channel.
+ * 
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ *
+ */
+public class HazelcastContinuousQueryMessageProducer extends HazelcastMessageProducer {
+
+	private String predicate;
+
+	@Override
+	protected void doStart() {
+		if (getDistributedObject() instanceof IMap) {
+			((IMap<?, ?>) getDistributedObject()).addEntryListener(
+					new HazelcastEntryListener(this), new SqlPredicate(this.predicate), true);
```

Comment:
For CQ Inbound Channel Adapter, it should be configurable because if **value** and **oldValue** are not needed, we can prevent unnecessary network traffic(specially, for "big data size" cases) by sending just other datas.

Also, in the light of your previous comment, i think we can create EntryEventMessagePayload  by covering *key*, *oldValue* and *value*. Also *entryEventType*, *member* and *name* can be set as Message Headers. WDYT?

Sample CQ Event : 
EntryEvent{entryEventType=UPDATED, member=Member [192.168.1.14]:5701 this, name='mapCQ', key=CQ0, oldValue=Employee{name='test', age='20'}, value=Employee{surname='test', age='30'}}